game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -84,9 +85,13 @@
   @Builder.Default boolean returnFire = true;
 
   /** When should this unit commit suicide on offense */
-  @Builder.Default Suicide suicideOnOffense = Suicide.NONE;
+  @Getter(value = AccessLevel.NONE)
+  @Builder.Default
+  Suicide suicideOnOffense = Suicide.NONE;
```

Comment:
Side-note, `Suicide` sounds like a value object, perhaps could be renamed to `SuicideCondition` or `UnitSuicideCondition`